client/js/ifml2code/client/elementrules/templates/system-event-vm.js.ejs

```diff
@@ -0,0 +1,51 @@
+/*jslint node: true, nomen: true */
+"use strict";
+
+var ko = require('knockout');
+
+function ViewModel(params) {
+    var self = this;
+
+    // TODO: Execution
+    /*
+      example:
+        //to trigger the event: params.fire;
+        document.addEventListener('keydown', params.fire);
+    */
+
+    self.context = params.context;
+
+    // THIS CAN BE REMOVED (BEGIN)
+    self.stop = function () {
+      // TODO: Execution
+      /*
+        example [continue]:
+          document.removeEventListener('keydown', params.fire);
+      */
+    };
+    // THIS CAN BE REMOVED (END)
+
+    self.trigger = function (id) {
+        self.context.events[id](self.context);
```

Comment:
Shouldn't you invoke the `fire`?
Invoking directly the event you cannot pass the data from ViewComponents

Reply:
What about renaming `events` to `navigations` for clearness?